lib/tlsutil/tlsutil.go

```diff
@@ -130,27 +128,36 @@ func (l *DowngradingListener) AcceptNoWrapTLS() (net.Conn, bool, error) {
 		return nil, false, err
 	}
 
-	br := bufio.NewReader(conn)
+	var first [1]byte
```

Comment:
(Making this a `[1]byte` instead of just a `byte` is more convenient as we can `Read()` into it)